### PR TITLE
feat(eap): Add an example endpoint that uses protobuf over http

### PR DESCRIPTION
**Pull request by colin-sentry**
Example usage:

```
import requests
from snuba.protobufs.FindTrace_pb2 import FindTraceRequest, FindTraceResponse

if __name__ == "__main__":
    req = FindTraceRequest(filters=[
        FindTraceRequest.Filter(
            span_with_attr_key_exists=FindTraceRequest.SpanWithAttrKeyExists(attr_key="release")
        ),
        FindTraceRequest.Filter(
            span_with_attr_key_equals_value=FindTraceRequest.SpanWithAttrKeyEqualsValue(attr_key="environment", attr_value="development")
        )
    ])
    http_resp = requests.post('http://localhost:1218/find_trace', data=req.SerializeToString())
    http_resp.raise_for_status()

    resp = FindTraceResponse()
    resp.ParseFromString(http_resp.content)
    print(resp.trace_uuids)
```

### CODE REVIEW

**Comment by evanh** [comment]
Why are we using protobuf here? Why not JSONSchema?

File: snuba/web/rpc/find_traces.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import asyncio
+from typing import List
+
+from snuba.clickhouse.formatter.nodes import FormattedQuery, StringNode
+from snuba.datasets.storages.factory import get_storage
+from snuba.datasets.storages.storage_key import StorageKey
+from snuba.protobufs.FindTrace_pb2 import FindTraceRequest, FindTraceResponse
```
[the remark this answers is not in the document]
Comment (colin-sentry) [reply snuba/web/rpc/find_traces.py:7]: I think a nice end state for the actual RPCs would be to use gRPC for everything, this is an intermediate state where we use the serialization but not the transport.

JSONSchema is fine, but certainly feels worse than grpc for this sort of thing

[the remark this answers is not in the document]
Comment (colin-sentry) [reply snuba/web/rpc/find_traces.py:7]: I've written some options in this notion doc if you are curious: 
https://www.notion.so/sentry/RPC-options-b4537d1a357e4a74be05be5b37ac6a82

File: snuba/web/rpc/find_traces.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        """
+    res = reader.execute(
+        FormattedQuery([StringNode(query)]),
+        robust=True,
+    )
+
+    return set(x["trace_id"] for x in res["data"])
+
+
+async def find_traces(req: FindTraceRequest, timer: Timer) -> FindTraceResponse:
```
Comment (volokluev) [inline snuba/web/rpc/find_traces.py:44]: NotImplementedError()

**Review by volokluev — APPROVED** [review]
since this is just an example this is fine, we'll probably change a lot of this but it's good to have an example

File: snuba/utils/constants.py [hunk 3]
```diff
@@ -6,6 +6,7 @@
 #: Metrics granularities for which a materialized view exist, in ascending order
 GRANULARITIES_AVAILABLE = (10, 60, 60 * 60, 24 * 60 * 60)
 
+
```
Comment (evanh) [inline snuba/utils/constants.py:9]: 💀

**Pull request merged**